tests/projects/test_projects.py

```diff
@@ -285,11 +285,12 @@ def test_run_async():
         ),
     ],
 )
-def test_conda_path(mock_env, expected_conda, expected_activate):
+def test_conda_path(mock_env, expected_conda, expected_activate, monkeypatch):
     """Verify that we correctly determine the path to conda executables"""
-    with mock.patch.dict("os.environ", mock_env, clear=True):
-        assert mlflow.utils.conda.get_conda_bin_executable("conda") == expected_conda
-        assert mlflow.utils.conda.get_conda_bin_executable("activate") == expected_activate
+    monkeypatch.delenvs(mock_env.keys(), raising=False)
```

Comment:
This still fails because mock_env only contains `CONDA_EXE` for the first test. 
```suggestion
    monkeypatch.delenvs(["CONDA_EXE", mlflow.utils.conda.MLFLOW_CONDA_HOME], raising=False)
```